File: src/main/kotlin/onl/tesseract/srp/skill/domain/model/crafting/CraftTask.java
package onl.tesseract.srp.skill.domain.model.crafting;


import java.time.Duration;

public record CraftTask(
        QueuedRecipe queuedRecipe,
        Duration unitDuration,
        Duration timeLeft
) {
    public CraftTask(
            QueuedRecipe queuedRecipe,
            Duration unitDuration
    ){
        this(queuedRecipe, unitDuration, unitDuration.multipliedBy(queuedRecipe.getQuantity()));
    }

    public CraftTask tick(Duration duration) {
        Duration newTimeLeft = timeLeft.minus(duration);
        if (newTimeLeft.isNegative()) {
            newTimeLeft = Duration.ZERO;
        }
        return new CraftTask(queuedRecipe, unitDuration, newTimeLeft);
    }

    public void decrementQuantity() {
        queuedRecipe.setQuantity(queuedRecipe.getQuantity() - 1);
    }
}

File: src/main/kotlin/onl/tesseract/srp/domain/skill/crafting/CraftTask.kt
package onl.tesseract.srp.domain.skill.crafting

import onl.tesseract.srp.domain.item.CustomItem
import kotlin.time.Duration

data class CraftTask(
    val queuedRecipe: QueuedRecipe,
    val done: MutableList<CustomItem>,
    val garbage: MutableList<CustomItem>,
    val unitDuration: Duration,
    val bonus: CraftingBonus = CraftingBonus(),
    var timeLeft : Duration = unitDuration.times(queuedRecipe.quantity)
) {

    fun getTotalTimeLeft(): Duration {
        return timeLeft
    }
}